experimental/cmd/inky/inky.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"image"
+	"image/png"
+	"log"
+	"os"
+
+	"periph.io/x/periph/conn/gpio/gpioreg"
+	"periph.io/x/periph/conn/spi/spireg"
+	"periph.io/x/periph/experimental/devices/inky"
+	"periph.io/x/periph/host"
+)
+
+var (
+	spiPort     = flag.String("spi", "SPI0.0", "Name or number of SPI port to open")
+	path        = flag.String("image", "", "Path to a png file to display on the inky")
+	dcPin       = flag.String("dc", "22", "Inky DC Pin")
+	resetPin    = flag.String("reset", "27", "Inky Reset Pin")
+	busyPin     = flag.String("busy", "17", "Inky Busy Pin")
+	model       = flag.String("model", "PHAT", "Inky model (PHAT or WHAT)")
+	modelColor  = flag.String("model-color", "red", "Inky model color (black, red or yellow)")
+	borderColor = flag.String("border-color", "black", "Border color (black, white, red or yellow)")
+)
+
+func parseModel(s string) inky.Model {
```

Comment:
What do you think of this? While this is way more code, the runtime behavior is more interesting, as it prints a proper message instead of crashing.

```
type modelType inky.Model

func (m *modelType) String() string {
	switch *m {
	case inky.PHAT:
		return "PHAT"
	case inky.WHAT:
		return "WHAT"
	default:
		return "Unknown"
	}
}

func (m *modelType) Set(s string) error {
	switch s {
	case "PHAT":
		*m = (modelType)(inky.PHAT)
	case "WHAT":
		*m = (modelType)(inky.WHAT)
	default:
		return fmt.Errorf("Unknown model %q: expected either PHAT or WHAT", s)
	}
	return nil
}

var model modelType

func main() {
	flag.Var(&model, "model", "PHAT", "Inky model (PHAT or WHAT)")
	(...)
}
```

It's also something we'd encourage to copy paste.

As a matter of fact I'd prefer putting these two methods directly on inky.Model instead, similar to https://godoc.org/periph.io/x/periph/conn/i2c#Addr. Then only the main() code above is needed here.

Same for color.

Reply:
Great ideas, done!